simulator/session_manager_test.go

```diff
@@ -224,3 +226,102 @@ func TestSessionManagerLoginExtension(t *testing.T) {
 		t.Error(err)
 	}
 }
+
+// Test WaitForUpdates against the PropertyCollector singleton.
+// Includes test for issue 1172, where a 2nd session use of the PropertyCollector singleton followed by Logout()
+// unregistered the 1st session's update listener as the PC singleton has the same moref regardless of session instance.
+func TestSessionManagerPropertyCollector(t *testing.T) {
+	ctx := context.Background()
+
+	m := VPX()
+
+	defer m.Remove()
+
+	err := m.Create()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	s := m.Service.NewServer()
+	defer s.Close()
+
+	c, err := govmomi.NewClient(ctx, s.URL, true)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	pc := property.DefaultCollector(c.Client)
+	root := object.NewRootFolder(c.Client)
+
+	filter := types.CreateFilter{
```

Comment:
This is from `govc object.collect -R create-filter-request.xml -O`, using the CreateFilter request from the trace in #1195